packages/blocks/person/src/App.tsx

```diff
@@ -1,38 +1,91 @@
-import React from "react";
-
+import React, { useMemo } from "react";
 import { BlockComponent } from "blockprotocol/react";
 
+import { MailIcon } from "./icons/MailIcon";
+import { LinkIcon } from "./icons/LinkIcon";
+
+import "./index.css";
+
 type AppProps = {
-  name?: string;
+  avatar?: string;
   email?: string;
+  employer?: {
+    name: string;
+    position: string;
+  };
+  link?: string;
+  name?: string;
 };
 
-export const App: BlockComponent<AppProps> = ({ name, email }) => {
+export const App: BlockComponent<AppProps> = ({
+  avatar,
+  employer,
+  email,
+  link,
+  name,
+}) => {
+  const { name: employerName, position } = employer ?? {};
+
+  const linkData = useMemo(() => {
+    if (!link) {
+      return null;
+    }
+
+    return new URL(link);
+  }, [link]);
+
   return (
-    <div
-      style={{
-        display: "flex",
-        fontFamily: "Helvetica, Arial, sans-serif",
-        fontSize: "19px",
-        fontWeight: 300,
-        lineHeight: "180%",
-      }}
-    >
-      <div
-        style={{
-          textAlign: "right",
-          marginRight: "15px",
-          fontWeight: 500,
-        }}
-      >
-        <div>Name: </div>
-        <div>Email: </div>
-      </div>
+    <div className="person-container">
+      {avatar && (
+        <img
+          alt={`${name ? `${name}'s ` : ""}avatar`}
+          className="avatar-desktop"
+          src={avatar}
+        />
+      )}
       <div>
-        <div>{name}</div>
-        <div>
-          <a href={`mailto:${email}`}>{email}</a>
+        <div className="person-name-container">
+          {avatar && (
+            <img
+              alt={`${name ? `${name}'s ` : ""}avatar`}
+              className="avatar-mobile"
+              src={avatar}
+            />
+          )}
+          <div className="person-name">{name}</div>
+          {employer && (
+            <div className="person-employer">
+              {position} <span className="person-employer-breaker">at</span>{" "}
+              {employerName}
+            </div>
+          )}
         </div>
+
+        {(email || link) && (
+          <>
+            <hr />
+            <div className="person-links-container">
+              {email && (
+                <div className="person-link">
+                  <MailIcon />
+
+                  <a href={`mailto:${email}`}>{email}</a>
+                </div>
+              )}
+
+              {link && (
+                <div className="person-link">
+                  <LinkIcon />
+
+                  <a href={link}>
```

Comment:
This needs to be sanitized to avoid things like `<a href="javascript:alert(2022)">click me</a>`

You could use [dompurify](https://github.com/cure53/DOMPurify):

```typescript
const safeAnchor = DOMPurify.sanitize(`<a href="${link}" />`, {
  ALLOWED_TAGS: {
    a: ["href"],
  },
});
```

Reply:
I feel like sanitization is something which the embedding application should worry about. 

Since we are only displaying input which is programmatically passed to our block, I don't think we should do input sanitization on our own, but allow the embedding blocks do decide which method they want to use.

Reply:
I agree that the embedding application should be worrying about input sanitization - but that blocks can also make some effort to minimise vulnerabilities. The trade-off here is the increased bundle size of the additional library, but it doesn't seem too bad in this case.

Reply:
Makes sense. Will implement this.

Reply:
The anchor generation code has been [added here](https://github.com/hashintel/hash/pull/350/files#diff-8cbd65a069e2bc56a0191f9ffb8af5950097076f29bfe2c188183b97c2f706adR32). [Using `dangerouslySetInnerHtml`](https://github.com/hashintel/hash/pull/350/files#diff-8cbd65a069e2bc56a0191f9ffb8af5950097076f29bfe2c188183b97c2f706adR104) to display the anchor.

That's the only way I saw online, let me know if I should use `react-html-parser` instead.